index.js

```diff
@@ -8,12 +8,22 @@ module.exports = function(content) {
 	this.cacheable && this.cacheable();
 	if(!this.emitFile) throw new Error("emitFile is required from module system");
 	var query = loaderUtils.parseQuery(this.query);
+	var configKey = query.config || "fileLoader";
+	var config = this.options[configKey] || {};
```

Comment:
`publicPath` cannot be defined in the query because the query is converted to and from JSON, so only JSON data types are supported (meaning the function gets cast to a string).